app/packages/core/src/components/ColorModal/GlobalSetting.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from "react";
+import { Divider, Slider } from "@mui/material";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { Check } from "@mui/icons-material";
+import { cloneDeep } from "lodash";
+import * as fos from "@fiftyone/state";
+
+import RadioGroup from "../Common/RadioGroup";
+import ColorPalette from "./colorPalette/ColorPalette";
+import Checkbox from "../Common/Checkbox";
+import { tempGlobalSetting } from "./utils";
+import {
+  ControlGroupWrapper,
+  LabelTitle,
+  SectionWrapper,
+} from "./ShareStyledDiv";
+
+const GlobalSetting: React.FC = ({}) => {
+  const [global, setGlobal] = useRecoilState(tempGlobalSetting);
+  const handleSliderChange = (event: Event, newValue: number | number[]) => {
+    setGlobal((s) => ({ ...cloneDeep(s), opacity: newValue as number }));
+  };
+  const colors = useRecoilValue(fos.coloring(false)).pool as string[];
+  const opacity = useRecoilValue(fos.alpha(false));
+  const colorBy = useRecoilValue(
+    fos.appConfigOption({ key: "colorBy", modal: false })
+  );
+  const useMulticolorKeypoints = useRecoilValue(
+    fos.appConfigOption({ key: "multicolorKeypoints", modal: false })
+  );
+  const showSkeleton = useRecoilValue(
+    fos.appConfigOption({ key: "showSkeletons", modal: false })
+  );
```

Comment:
nit: same here, similar code across at least 3/4 files, would be nice to refactor these into a hook